### PR TITLE
feat(web/applications): add key dates page (boas-684)

### PR DESCRIPTION
**Pull request by lutamKC**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/testing/applications/factory/util.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const createUniqueRandomNumberFromSeed = (min, max, seed) => {
 	const randomFloat =
 		Number.parseInt([...date.getTime().toString()].reverse().slice(-3).join(''), 10) / 1000;
 
-	return Math.floor(randomFloat * max) + min;
+	return Math.floor(randomFloat * (max - min)) + min;
```
Comment (lutamKC) [inline apps/web/testing/applications/factory/util.js:32]: the reason why kind of every snapshot has changed.

File: apps/web/src/styles/7-components/ListMenu/ListMenu.scss
Comment (lutamKC) [inline apps/web/src/styles/7-components/ListMenu/ListMenu.scss:1]: refactored as other files in the format /components/name-of-style.scss

File: apps/web/src/server/views/applications/case/layouts/applications-case-layout.njk [hunk 2]
```diff
@@ -41,13 +41,13 @@
 {% endblock %}
 
 {% block pageContent %}
-	<div class="govuk-grid-row">
+	<div class="govuk-grid-row display--flex">
 		{% block sectionMenu %}
-			<div class="govuk-grid-column-one-third">
+			<div class="pins-list-menu-container">
```
Comment (lutamKC) [inline apps/web/src/server/views/applications/case/layouts/applications-case-layout.njk:46]: side menu of case pages (all of them), was shrinked. From 1/3 of the page (standard govuk width), to only 220px.

this is one of the reason of the many updated snapshots

File: apps/web/src/server/lib/nunjucks-filters/key-dates-property.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/** @type {Record<string, string>} */
+const propertyToName = {
```
Comment (lutamKC) [inline apps/web/src/server/lib/nunjucks-filters/key-dates-property.js:2]: I hope the mapping is correct. 

Thanks to @chrisdalykinandcarta  the names were straightforwarding, but I could have done some mistake anyway.

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js [hunk 4]
```diff
@@ -87,6 +87,7 @@ export async function viewApplicationsCaseTimetableList(_, response) {
 	response.render(`applications/case-timetable/timetable-list`, {
 		timetableItems: timetableItemsViewData,
 		publishedStatus: examinationTimetable?.published,
+		selectedPageType: 'examination-timetable',
```
Comment (lutamKC) [inline apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:90]: unrelated fix: highlight the menu item when on the page

File: apps/api/src/server/utils/mapping/map-key-dates.js [hunk 5]
```diff
@@ -39,24 +39,24 @@ export const mapRequestToKeyDates = ({
...
-			...mapValues(
-				pick(
-					keyDates,
-					preApplicationDateNames.filter((key) => key != 'submissionAtPublished')
-				),
-				mapDateToUnixTimestamp
-			),
-			submissionAtPublished: keyDates.submissionAtPublished
-		},
+		preApplication: mapValues(pick(keyDates, preApplicationDateNames), mapPreApplicationDates),
```
Comment (lutamKC) [inline apps/api/src/server/utils/mapping/map-key-dates.js:59]: design expects properties to be in a specific order, at least in this section ("Anticipated submission date published" should be right before "Anticipated submission date internal").

in this way we only map the dates as it was before (so we exclude submissionAtPuslibhshed: string), but keeping the same order of the properties

Comment (chrisdalykinandcarta) [reply apps/api/src/server/utils/mapping/map-key-dates.js:59]: Makes sense, thanks for explanation! Also I think your code is more readable than mine anyway.

**Review by francescobrenci-AR — APPROVED** [review]
